src/main/java/org/openstreetmap/atlas/utilities/filters/AtlasEntityPolygonsFilter.java

```diff
@@ -0,0 +1,168 @@
+package org.openstreetmap.atlas.utilities.filters;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.openstreetmap.atlas.geography.Polygon;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasEntity;
+import org.openstreetmap.atlas.geography.converters.PolygonStringConverter;
+import org.openstreetmap.atlas.geography.converters.WkbPolygonConverter;
+import org.openstreetmap.atlas.geography.converters.WktPolygonConverter;
+import org.openstreetmap.atlas.geography.index.QuadTree;
+import org.openstreetmap.atlas.streaming.readers.GeoJsonReader;
+import org.openstreetmap.atlas.streaming.readers.json.serializers.PropertiesLocated;
+import org.openstreetmap.atlas.streaming.resource.StringResource;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.openstreetmap.atlas.utilities.conversion.StringConverter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.vividsolutions.jts.io.WKBReader;
+
+/**
+ * Configurable {@link AtlasEntity} filter that passes through anything that intersects with include
```

Comment:
If I wanted to extend this functionality to be slightly different - say instead of intersecting given polygons, want fully enclosed within given polygons - this means I'd have to change this core class quite a bit. What are your thoughts on making this more generic - and allowing you to plug any kind of filters/predicates in given some set of polygons? I think we talked offline about TaggableFilter being generic enough to handle any tag combination. In my mind, this would be a lot more powerful if it could handle any kind of entity inclusion/exclusion rules.

Reply:
Okay, can we go over of the dimensions? So we want to be able to choose which of the following:
<li>Include versus exclude dominant</li>
<li>Intersects versus fully geometrically encloses </li>
<li> Overlapping allowed versus disallowed (And that effect of the dominance)</li>
<li>Settings specific to polygon type or generic</li>

Reply:
Also, should we make these setting file configurable or provided by the user depending on filter use case?

Reply:
I would lean towards... we want as much provided by the user as possible. They can specify their own filters and sets of polygons. Making it configurable is nice, but they can do that in the client code. This way we don't have to deal with updating various configurations in this class.